lib/core/util.js

```diff
@@ -244,13 +244,32 @@ function validateHandler (handler, method, upgrade) {
 // A body is disturbed if it has been read from and it cannot
 // be re-used without losing state or data.
 function isDisturbed (body) {
+  if (!body) {
+    return false
+  }
+
+  if (stream.isDisturbed) {
+    return stream.isDisturbed(body)
+  }
+
   const state = body && body._readableState
-  return !!(body && (
-    (stream.isDisturbed && stream.isDisturbed(body)) ||
+  return !!(
     body[kBodyUsed] ||
     body.readableDidRead || (state && state.dataEmitted) ||
-    isAborted(body)
-  ))
+    isReadableAborted(body)
```

Comment:
This could be `isDestroyed`, no? Is there a reason why we check `endEmitted`?

Reply:
Yes. If it is destroyed before emitting end then and only the is it aborted.